app/src/main/java/org/xbmc/kore/ui/sections/remote/PlaylistFragment.java

```diff
@@ -223,32 +224,38 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
-    private boolean refreshingPlaylist;
-    private void refreshPlaylist(GetPlaylist getPlaylist) {
-        if (refreshingPlaylist)
```

Comment:
Protecting this check and assignment with a `lock` doesn't also fix the possible race condition?

Reply:
No, the problem occurs when the playlist tests add multiple playlists in the setup fase before each test. When I implemented #618 I only implemented tests using a single playlist. Afterwards I added some additional tests to test more scenario's which caused the race condition to arise. Probably not something that will/can happen in real usage but is at least needed to make the playlist tests work.